pkg/alertmanager/amcfg.go

```diff
@@ -313,13 +354,21 @@ func (cb *configBuilder) addAlertmanagerConfigs(ctx context.Context, amConfigs m
 			continue
 		}
 
+		// verifiy if it is the last amConfig in the namespace
+		if i+1 < len(amConfigIdentifiers) {
+			nextAmConfig := amConfigs[amConfigIdentifiers[i+1]]
+			cb.continueStrategy.isLastRoute = nextAmConfig.Namespace != crKey.Namespace
+		} else {
+			cb.continueStrategy.isLastRoute = true
+		}
+
 		subRoutes = append(subRoutes,
 			cb.enforcer.processRoute(
 				crKey,
 				cb.convertRoute(
 					amConfigs[amConfigIdentifier].Spec.Route,
 					crKey,
-				),
+				), &cb.continueStrategy,
```

Comment:
hmm it would be simpler to pass a bool stating whether the route is last or not?